hazelcast/src/main/java/com/hazelcast/query/impl/AndResultSet.java

```diff
@@ -161,6 +153,28 @@ public void remove() {
 
     @Override
     public int size() {
-        return setSmallest.size();
+        if (cachedSize == SIZE_UNINITIALIZED) {
+            int calculatedSize = 0;
+            for (Iterator<QueryableEntry> it = iterator(); it.hasNext(); it.next()) {
+                calculatedSize++;
+            }
+            cachedSize = calculatedSize;
+        }
+        return cachedSize;
     }
+
+    /**
+     * @return returns estimated size without calculating the full result set in full-result scan.
+     */
+    public int estimatedSize() {
```

Comment:
if `estimatedSize()` is called on this result set before `size() `,  is there a possibility that returned value from `size()` is wrong because of the _size-caching_?